src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -706,13 +704,13 @@ ParsedQuery Visitor::visit(Parser::ModifyContext* ctx) {
 // ____________________________________________________________________________________
 vector<SparqlTripleSimpleWithGraph> Visitor::visit(
     Parser::DeleteClauseContext* ctx) {
-  return visit(ctx->quadPattern());
+  return visit(ctx->quadPattern()).getQuads();
```

Comment:
How much do you gain if you refactor more of the functions that currently return
`vector<SparqlTripleSimpleWithGraph>` s.t. they return `Quads`?
Maybe you save calls to `getQuads()` and get expressive datatypes more often.

For the whole registering you can also do this on the quads directly (implement a function "for all variables" or "get all Variables" etc.

Reply:
More expressive and easier to work with datatypes would be a reason to do this. There are no redundant calls to `getQuads`.